src/libraries/System.Private.CoreLib/src/System/DefaultBinder.cs

```diff
@@ -769,12 +769,12 @@ public sealed override void ReorderArgumentArray(ref object?[] args, object stat
 
         // Return any exact bindings that may exist. (This method is not defined on the
         //  Binder and is used by RuntimeType.)
-        public static MethodBase? ExactBinding(MethodBase[] match, Type[] types, ParameterModifier[]? modifiers)
+        public static RuntimeConstructorInfo? ExactBinding(RuntimeConstructorInfo[] match, Type[] types, ParameterModifier[]? modifiers)
```

Comment:
Using RuntimeConstructorInfo will make sharing of this code problematic with NativeAOT. 

DefaultBinder is a super slow path. An extra casts here won't make difference.